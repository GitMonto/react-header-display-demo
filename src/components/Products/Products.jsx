import PropTypes from "prop-types";

function createProductLIs(productsData) {
  return productsData.map((product) => (
    <li
      key={product.name}
      className="mx-auto flex w-48 flex-col items-center pt-11"
    >
      {product.name} - {product.price}
    </li>
  ));
}

export default function Products({ products }) {
  return <ul>{createProductLIs(products)}</ul>;
}

Products.propTypes = {
  products: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      price: PropTypes.string.isRequired,
    })
  ).isRequired,
};
