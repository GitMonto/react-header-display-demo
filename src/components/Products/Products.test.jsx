// Test this component with dummy/mock data
import renderer from "react-test-renderer";
import Products from "./Products";

const products = [
  {
    price: "$49.99",
    name: "Test 1",
  },
  {
    price: "$9.99",
    name: "Test 2",
  },
  {
    price: "$29.99",
    name: "Test 3",
  },
];

// TODO: Write a snapshot test and visually inspect it to insure that it is rendering correctly
it("renders without crashing", () => {
  const rendered = renderer.create(<Products products={products} />);
  expect(rendered).toMatchSnapshot();
});
