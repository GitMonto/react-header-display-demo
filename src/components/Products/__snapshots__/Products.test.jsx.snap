// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<ul>
  <li
    className="mx-auto flex w-48 flex-col items-center pt-11"
  >
    Test 1
     - 
    $49.99
  </li>
  <li
    className="mx-auto flex w-48 flex-col items-center pt-11"
  >
    Test 2
     - 
    $9.99
  </li>
  <li
    className="mx-auto flex w-48 flex-col items-center pt-11"
  >
    Test 3
     - 
    $29.99
  </li>
</ul>
`;
